From d9a9db7dafde10809a05a4e5116d9e02feb05dc5 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Fri, 19 Jan 2018 20:04:29 -0700
Subject: [PATCH 2/2] Revert "msm8974-common: Use QTI HIDL power HAL"

This reverts commit e9559143366bd64843aa9267a419fe2bfbf83b8b.

* As it turns out, the brand new HILD power HAL isn't ready, at
  least not for legacy devices like msm8974. Symptoms were that
  after waking from sleep, cores 1-3 are not brought back online
  consistently.
* Let's go back to the old thing, at least for now.

Change-Id: Ica2ddbeb42635662167efe0b48f0fe4051591b26
---
 BoardConfigCommon.mk | 5 -----
 manifest.xml         | 2 +-
 msm8974.mk           | 3 ++-
 3 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 43beb88..1a1c944 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -67,11 +67,6 @@ DEVICE_MATRIX_FILE := device/samsung/msm8974-common/compatibility_matrix.xml
 TARGET_INIT_VENDOR_LIB := libinit_msm8974
 TARGET_RECOVERY_DEVICE_MODULES := libinit_msm8974
 
-# Power
-TARGET_HAS_LEGACY_POWER_STATS := true
-TARGET_HAS_NO_WIFI_STATS := true
-TARGET_USES_INTERACTION_BOOST := true
-
 # Time services
 BOARD_USES_QC_TIME_SERVICES := true
 
diff --git a/manifest.xml b/manifest.xml
index fdffd97..d38a208 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -116,7 +116,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.power</name>
-        <transport>hwbinder</transport>
+        <transport arch="32">passthrough</transport>
         <version>1.0</version>
         <interface>
             <name>IPower</name>
diff --git a/msm8974.mk b/msm8974.mk
index bb6710d..c8cb980 100644
--- a/msm8974.mk
+++ b/msm8974.mk
@@ -142,7 +142,8 @@ PRODUCT_PACKAGES += \
 
 # Power HAL
 PRODUCT_PACKAGES += \
-    android.hardware.power@1.0-service-qti
+    android.hardware.power@1.0-impl \
+    power.msm8974
 
 # RenderScript HAL
 PRODUCT_PACKAGES += \
-- 
2.7.4

